src/coreclr/nativeaot/System.Private.CoreLib/src/System/Activator.NativeAot.cs

```diff
@@ -49,7 +49,7 @@ public static partial class Activator
             try
             {
                 // Call the default constructor on the allocated instance.
-                if (RuntimeHelpers.IsReference<T>())
+                if (!typeof(T).IsValueType)
```

Comment:
@MichalStrehovsky Can we assume that `typeof(T).IsValueType` is always recognized as an intrinsic by the toolchain and that `typeof(T)` won't introduce any undesirable additional reflection dependencies?

Reply:
Since this is in activator implementation, I'm not too worried about possibility of more reflection dependencies - we're already allocating the type and it better be seen as reflected on and allocated.

But if the PR description has a claim of "Should improve codegen", I would want to see that claim backed up by disassembly of before/after code for a value type and for a ref type to ensure there's no surprise regression.